dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESReadOnlyMonitorTest.java

```diff
@@ -0,0 +1,315 @@
+     * When: If the LIVE and WORKING current index are not read only
+     * Should: not sent large message
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldNotSendLargeMessage() throws DotDataException, IOException {
+        final String message = "message";
+
+        final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+
+        setReadOnly(indiciesInfo.getWorking(), false);
+        setReadOnly(indiciesInfo.getLive(), false);
+
+        esReadOnlyMonitor.start(message);
+
+        verify(systemMessageEventUtilMock, never()).pushLargeMessage(any(), any());
+
+        assertEquals(false, ElasticsearchUtil.isAnyReadOnly(indiciesInfo.getWorking(), indiciesInfo.getLive()));
+    }
+
+    /**
+     * Method to Test: {@link ESReadOnlyMonitor#start(String)}
+     * When: If any of LIVE and WORKING current index are read only
+     * Should: sent large message
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSendLargeMessage() throws DotDataException, DotSecurityException, InterruptedException, IOException {
+        final String message = "message";
+
+        final IndiciesInfo indiciesInfo = APILocator.getIndiciesAPI().loadIndicies();
+
+        final Role adminRole = mock(Role.class);
+        when(roleAPIMock.loadCMSAdminRole()).thenReturn(adminRole);
+
+        final User user = mock(User.class);
+        when(user.getUserId()).thenReturn("1");
+
+        when(roleAPIMock.findUsersForRole(adminRole)).thenReturn(list(user));
+
+        try {
+            setReadOnly(indiciesInfo.getWorking(), true);
+            setReadOnly(indiciesInfo.getLive(), false);
+
+            esReadOnlyMonitor.start(message);
+
+            Thread.sleep(100);
+
+            checkLargeMessageSent(user);
+            assertEquals(false, ElasticsearchUtil.isAnyReadOnly(indiciesInfo.getWorking(), indiciesInfo.getLive()));
+        } finally {
+            setReadOnly(indiciesInfo.getWorking(), false);
+        }
+    }
+
+    /**
+     * Method to Test: {@link ESReadOnlyMonitor#start()}
+     * When: If the cluster is read only
+     * Should: sent message
+     *
+     * @throws DotDataException
+     */
+    @Test
+    public void shouldSendLargeMessageIfTheClusterIsInReadOnly() throws DotDataException, DotSecurityException, InterruptedException, IOException {
+        final String message = "message";
+
+        final Role adminRole = mock(Role.class);
+        when(roleAPIMock.loadCMSAdminRole()).thenReturn(adminRole);
+
+        final User user = mock(User.class);
+        when(user.getUserId()).thenReturn("1");
+
+        when(roleAPIMock.findUsersForRole(adminRole)).thenReturn(list(user));
+
+        try {
+            setClusterAsReadOnly(true);
+
+            esReadOnlyMonitor.start(message);
+
+            Thread.sleep(100);
+
+            checkClusterLargeMessageSent(user);
+            assertEquals(false, ElasticsearchUtil.isClusterInReadOnlyMode());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use assertTrue(x)/assertFalse(x) instead of assertEquals(true, x)/assertEquals(false, x) or assertEquals(Boolean.TRUE, x)/assertEquals(Boolean.FALSE, x).](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5642172)